### DISCUSSION TITLE
NFC Check-in & Check-Out

### DISCUSSION THREAD

**Discussion by epe** [kromitgmbh/titra#205] [Ideas]
Hi
How would we go about to enable NFC Card Check in & Check out? 
I would suggest an ESP Microcontroller with NFC Reader that sends the cards ID or an API Request to titra?

Also we have barcode scanners at most PCs in the workshop, so we could also use that possibly..
Best regards!

**Comment by faburem** [comment]
Hello,
yes you can use the titra API for this e.g. with the start and stop timer API endpoints:
https://kromitgmbh.github.io/titra/#api-TimeEntry-startTimer
https://kromitgmbh.github.io/titra/#api-TimeEntry-stopTimer

The titra API is using the API token from the users settings page:
<img width="1428" alt="image" src="https://github.com/kromitgmbh/titra/assets/11456790/03e209bb-028d-4095-8de9-4b83d35672be">

I think this will be the main challenge, that you need a dedicated service to map the tokens to the NFC Card IDs.
OR thinking about it more, you could update the titra API token to match the NFC Card IDs - that should make the API call very straight forward.
